deepchem/molnet/load_function/delaney_datasets.py

```diff
@@ -34,95 +57,38 @@ def load_delaney(featurizer='ECFP',
-
-    if featurizer == "smiles2img":
-      img_spec = kwargs.get("img_spec", "std")
-      save_folder = os.path.join(save_folder, img_spec)
-    save_folder = os.path.join(save_folder, str(split))
-
-  dataset_file = os.path.join(data_dir, "delaney-processed.csv")
-
-  if not os.path.exists(dataset_file):
-    deepchem.utils.data_utils.download_url(url=DELANEY_URL, dest_dir=data_dir)
-
-  delaney_tasks = ['measured log solubility in mols per litre']
-  if reload:
-    loaded, all_dataset, transformers = deepchem.utils.data_utils.load_dataset_from_disk(
-        save_folder)
-    if loaded:
-      return delaney_tasks, all_dataset, transformers
-
-  if featurizer == 'ECFP':
-    featurizer = deepchem.feat.CircularFingerprint(size=1024)
-  elif featurizer == 'GraphConv':
-    featurizer = deepchem.feat.ConvMolFeaturizer()
-  elif featurizer == 'Weave':
-    featurizer = deepchem.feat.WeaveFeaturizer()
-  elif featurizer == 'Raw':
-    featurizer = deepchem.feat.RawFeaturizer()
-  elif featurizer == "smiles2img":
-    img_spec = kwargs.get("img_spec", "std")
-    img_size = kwargs.get("img_size", 80)
-    res = kwargs.get("res", 0.5)
-    featurizer = deepchem.feat.SmilesToImage(
-        img_size=img_size, img_spec=img_spec, res=res)
-
-  loader = deepchem.data.CSVLoader(
-      tasks=delaney_tasks, feature_field="smiles", featurizer=featurizer)
-  dataset = loader.create_dataset(dataset_file, shard_size=8192)
-
-  if split is None:
-    transformers = [
-        deepchem.trans.NormalizationTransformer(
-            transform_y=True, dataset=dataset, move_mean=move_mean)
-    ]
-
-    logger.info("Split is None, about to transform data")
-    for transformer in transformers:
-      dataset = transformer.transform(dataset)
-
-    return delaney_tasks, (dataset, None, None), transformers
-
-  splitters = {
-      'index': deepchem.splits.IndexSplitter(),
-      'random': deepchem.splits.RandomSplitter(),
-      'scaffold': deepchem.splits.ScaffoldSplitter(),
-      'stratified': deepchem.splits.SingletaskStratifiedSplitter()
-  }
-  splitter = splitters[split]
-  logger.info("About to split dataset with {} splitter.".format(split))
-  train, valid, test = splitter.train_valid_test_split(dataset)
-
-  transformers = [
-      deepchem.trans.NormalizationTransformer(
-          transform_y=True, dataset=train, move_mean=move_mean)
-  ]
-
-  logger.info("About to transform data.")
-  for transformer in transformers:
-    train = transformer.transform(train)
-    valid = transformer.transform(valid)
-    test = transformer.transform(test)
-
-  if reload:
-    deepchem.utils.data_utils.save_dataset_to_disk(save_folder, train, valid,
-                                                   test, transformers)
-  return delaney_tasks, (train, valid, test), transformers
+  loader = _DelaneyLoader(
+      featurizer, splitter, data_dir, save_dir, move_mean=move_mean, **kwargs)
+  featurizer_name = str(loader.featurizer)
```

Comment:
Should we just make a helper function to get the save folder name? The "_mean_unmoved" makes it a little awkward for Delaney  but this feels like a common pattern

Reply:
We really should have tests for all molnet loader functions.  But running all of them will take several hours at least and involve downloading many GB of data.  Figuring out a good strategy for testing molnet is beyond the scope of this PR.

Reply:
I tried to come up with a generic way for setting the folder name, but couldn't think of anything clean.  The code here to assemble the path is five lines, and three of those five lines are in some way specific to this dataset.